cyrasterize/shader.pyx

```diff
@@ -0,0 +1,92 @@
+from libcpp cimport bool
+from c_opengl cimport *
+import logging as log
+
+from c_opengl_debug cimport *
+import sys
+
+__all__ = ['VertexShader', 'FragmentShader']
+
+cdef class ShaderSource:
+    # Most of this code comes from kivy
+    # https://github.com/kivy/kivy/blob/master/kivy/graphics/shader.pyx
```

Comment:
I had a look at Kivy source at the beginning as it was quite close to what we wanted by reimplemented most stuff from scratch as I found their code to be either a bit too complex or had a lot of features we didn't really need. I rephrased the comment in 2b038b6dc20b22072124db8a2afab7dbfca97f71.